highcharts-angular/src/lib/highcharts-chart.directive.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  Directive,
+  DestroyRef,
+  afterRenderEffect,
+  ElementRef,
+  inject,
+  input,
+  linkedSignal,
+  model,
+  output,
+  OutputEmitterRef,
+  untracked,
+  computed
+} from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
+import { HighchartsChartService } from './highcharts-chart.service';
+import { HIGHCHARTS_CONFIG } from './highcharts-chart.token';
+import { Chart } from './types';
+
+
+@Directive({
+  selector: '[highcharts-chart]',
+})
+export class HighchartsChartDirective {
+  /**
+   * Highcharts library or Highcharts ESM module.
+   * @deprecated
+   */
+  Highcharts = input<Chart['highcharts']>();
+
+  /**
+   * Type of the chart constructor.
```

Comment:
Add what choice do we have.

Reply:
@karolkolodziej I have added ChartConstructorType to provide all constructorTypes, please let me know if I am missing some.